### PR TITLE
Simplify approval tests

### PR DESCRIPTION
**Pull request by agluszak**
Instead of an external library use the same mechanism we have for bsp4j and bsp4s.

### CODE REVIEW

**Review by ckipp01 — APPROVED** [review]
Good idea, this make sense. 👍🏼 Looks like there are some CI failures though around formatting, but I like the general idea.

**Review by adpi2 — APPROVED** [review]
LGTM

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -68,18 +68,9 @@ need to be replaced with the following 2 steps:
...
-In directory `codegen/src/test/scala` there will appear
-files `<testname>.<generated doc>.received.txt` for each generated doc which differs
-from the previously approved version. Approving means comparing `<file>.received.txt`
-with `<file>.approved.txt` and introducing changes to the `<file>.approved.txt` one.
-
-When you run `sbt test` again the tests should pass and `<file>.received.txt`
-will be removed.
+Upon committing changes to the protocol specification, a test is run to ensure that the generated
+protocol specification is identical to the committed one.
+If the test fails, please run `sbt generate` and commit the changes.
```
Comment (jastice) [inline CONTRIBUTING.md:73]: Isn't there some case where one would like to fix something?

Comment (agluszak) [reply CONTRIBUTING.md:73]: What do you mean?

Comment (jastice) [reply CONTRIBUTING.md:73]: I mean, in which cases should I do something other than just running generate and commit?

Comment (jastice) [reply CONTRIBUTING.md:73]: Never ever? Or just when nothing changes in the spec but the test breaks anyway?

Comment (agluszak) [reply CONTRIBUTING.md:73]: I'm not sure what you're referring to

Comment (agluszak) [reply CONTRIBUTING.md:73]: If nothing changes in the spec, but the test breaks anyway, that's a bug

**Pull request merged**